wire-ios/Wire-iOS/Sources/Analytics/TrackingManager.swift

```diff
@@ -22,24 +22,47 @@ import WireCommonComponents
 import WireSyncEngine
 
 final class TrackingManager: NSObject, TrackingInterface {
+
     private let flowManagerObserver: NSObjectProtocol
+    private let sessionManager: SessionManager
+
+    init(sessionManager: SessionManager) {
+        self.sessionManager = sessionManager
 
-    private override init() {
         AVSFlowManager.getInstance()?.setEnableMetrics(!ExtensionSettings.shared.disableAnalyticsSharing)
 
-        flowManagerObserver = NotificationCenter.default.addObserver(forName: FlowManager.AVSFlowManagerCreatedNotification, object: nil, queue: OperationQueue.main, using: { _ in
-            AVSFlowManager.getInstance()?.setEnableMetrics(!ExtensionSettings.shared.disableAnalyticsSharing)
-        })
-    }
+        flowManagerObserver = NotificationCenter.default.addObserver(
+            forName: FlowManager.AVSFlowManagerCreatedNotification,
+            object: nil,
+            queue: OperationQueue.main,
+            using: { _ in
+                AVSFlowManager.getInstance()?.setEnableMetrics(!ExtensionSettings.shared.disableAnalyticsSharing)
+            }
+        )
 
-    static let shared = TrackingManager()
+        super.init()
+    }
 
     var disableAnalyticsSharing: Bool {
-        get { ExtensionSettings.shared.disableAnalyticsSharing }
+        get {
+            ExtensionSettings.shared.disableAnalyticsSharing
+        }
         set {
-            Analytics.shared?.isOptedOut = newValue
-            AVSFlowManager.getInstance()?.setEnableMetrics(!newValue)
             ExtensionSettings.shared.disableAnalyticsSharing = newValue
+            AVSFlowManager.getInstance()?.setEnableMetrics(!newValue)
+
+            do {
+                if newValue {
+                    let disableUseCase = try sessionManager.makeDisableAnalyticsSharingUseCase()
+                    disableUseCase.invoke()
+                } else {
+                    let enableUseCase = try sessionManager.makeEnableAnalyticsSharingUseCase()
+                    enableUseCase.invoke()
+                }
+            } catch {
+                print("Failed to toggle analytics sharing: \(error)")
```

Comment:
This should be a log, and in case it fails, we need to reset the setting.

Reply:
Correct! Fixed it here: https://github.com/wireapp/wire-ios/commit/1bd1e64892df204e1cfd17260a6b1ba72a15e3a2